File: php/features.php
<?php
declare(strict_types=1);

// Kota SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KotaFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KotaBaseFeature();
            case "test":
                return new KotaTestFeature();
            default:
                return new KotaBaseFeature();
        }
    }
}

File: py/features.py
# Kota SDK feature factory

from feature.base_feature import KotaBaseFeature
from feature.test_feature import KotaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KotaBaseFeature(),
        "test": lambda: KotaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
